server/app/models/SchedulingModel.ts

```diff
@@ -6,11 +6,13 @@ export interface IScheduling extends Document {
   quantity: number;
   cost: number;
   operatingTime: Date;
+  machineName: string;
 }
 
 export const SchedulingSchema: Schema = new Schema({
   partType: { type: Schema.Types.String, required: true, immutable: true },
   quantity: { type: Schema.Types.Number, required: true, min: 1 },
   cost: { type: Schema.Types.Number, required: true, min: 1 },
-  operatingTime: { type: Schema.Types.Date, default: Date.now() },
+  operatingTime: { type: Schema.Types.Date, required: true },
```

Comment:
should we add a frequency such as daily, weekly, monthly,.... Also, should we have a start and end date for the machine scheduling?

Reply:
yeaa thats a good idea.

I removed operatingTime so that we only have StartTime and EndTime
I added frequency as well.

I dont know if duration is necessary now for machine but i dont know what to put for it

Reply:
is it better to have both `StartDate` , `EndDate`,  ` StartTime` and `EndTime`? Similar to the process of scheduling an event on Calendar app in Iphone. What do you think? 

If we have start/end Time, then we don't need duration